### PR TITLE
Improve clarity of editing session messages

### LINKED ISSUE #12167: Translation concern in Wagtail 6.2 save dialog
## Issue

Wagtail 6.2rc1 introduces a confirmation dialog when you are about to override changes made in a different window.

The text in this dialog uses a [demonstrative pronoun](https://www.grammarly.com/blog/demonstrative-pronouns/) (this/that/these/those) in combination with a templated, unknown, noun. 

https://github.com/wagtail/wagtail/blob/6c11616bd6cb3803e00fb03c10d48b0a97b2442a/wagtail/admin/templates/wagtailadmin/shared/editing_sessions/list.html#L114

This poses an issue for Dutch (and likely other languages) because the correct translation of _this_ differs depending on the gender of `model_name`. Unfortunately, _this_ (😉)  is not a solved problem in gettext.

## Proposal

I propose we remove the reference to a model from the dialog text. The surrounding context on the page should be plenty clear enough, the message will not lose any clarity.

image

```diff
- {{ user_name }} has saved a newer version of this {{ model_name }}
+ {{ user_name }} has saved a newer version

- You have saved a newer version of this {{ model_name }} in another session
+ You have saved a newer version in another session
```

---

More generally speaking, use of demonstrative pronouns in combination with some unknown noun is a recurring issue in Wagtail. Often, there are ways to creatively rewrite a sentence to avoid use demonstrative pronouns but Transifex makes this hard because it marks strings as invalid if you don't include the placeholder value for the noun, limiting translator options.

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #12167

### CODE REVIEW

File: wagtail/admin/templates/wagtailadmin/shared/editing_sessions/list.html [hunk 1]
```diff
@@ -106,12 +106,12 @@
 {% if sessions.0.revision_id %}
     <template data-controller="w-teleport" data-w-teleport-target-value="#title-text-w-overwrite-changes-dialog" data-w-teleport-reset-value="true">
         {% if current_session.user == sessions.0.user %}
-            {% blocktranslate trimmed with model_name=content_type.name asvar someone_has_saved_message %}
-                You have saved a newer version of this {{ model_name }} in another session
+            {% blocktranslate trimmed asvar someone_has_saved_message %}
+                You have saved a newer version in another window
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/shared/editing_sessions/list.html:110]:
```suggestion
                You saved a new version in another window
```

I wonder if we should reuse the exact same wording to simplify further?

**Review by thibaudcolas — COMMENTED** [review]
Code changes looking good, haven’t tested yet

File: wagtail/admin/templates/wagtailadmin/shared/editing_sessions/list.html (hunk 1 above)
Comment (laymonage) [reply wagtail/admin/templates/wagtailadmin/shared/editing_sessions/list.html]: Yeah, I don't see a reason why it needs to be different to the one in the popup. I've amended the commit to simplify this as well. Thanks!

**Review by thibaudcolas — APPROVED** [review]
Thank you!

**Comment by Stormheg** [comment]
🫶

**Pull request merged**